### DISCUSSION TITLE
Provider not reloading after BottomSheet is closed prematurely

### DISCUSSION THREAD

**Discussion by furkansoylemez** [rrousselGit/riverpod#2660] [Q&A] [answered]
Hello, I have a provider that I use to fetch and display data:

```
 part 'favorites_provider.g.dart';
@riverpod
Future<List<UserFavoritedChargePointDto>> favorites(FavoritesRef ref) async {
  final response = await ref.read(getFavoritesUseCaseProvider).call();
  return response;
}
```

I only use this provider on this page:

```
class FavoritesModal extends ConsumerWidget {
  const FavoritesModal({
    super.key,
  });

  @override
  Widget build(BuildContext context, WidgetRef ref) {
    final favorites = ref.watch(favoritesProvider);

    ref.listen(favoritesProvider, (_, state) {
      context.showSnackBarOnErrorState(state);
    });

    return Column(… in somewhere this page i use favorites.when to manage widgets also
```

This page is displayed in a BottomSheet like this:

```
void _showFavoritesBottomSheet() {
    context.showBottomSheet(
      height: 690.h,
      child: const FavoritesModal(),
    );
  }
```

When I open the BottomSheet, the provider fetches the data as expected and if there's an error, it's caught in the listen callback and a snackbar is displayed. If I wait for the snackbar to disappear and then close the BottomSheet, the next time I open it, it fetches the data again. This is the behavior I want.

However, if I close the BottomSheet before the snackbar disappears, the next time I open it, it doesn't fetch the data. I see it remains in the error state. Since I'm not listening to this provider anywhere else and I close the BottomSheet, I expect the state to reset. The same problem persists, even if I wait for the snackbar to disappear to reopen after closing it while the snackbar is visible.

What am I missing here?

**Comment by rrousselGit** [comment]
Like I mentioned in the issue, yu'll have to share a full example. Not much that can be done otherwise.

**Comment by furkansoylemez** [comment, marked as answer]
A workaround:

The problem I faced was that if I closed the bottomSheet while the SnackBar was visible, the provider would not dispose. However, if I closed it after the SnackBar disappeared, it would correctly dispose. To prevent this situation, I set the property of the Flushbar I use to show the SnackBar as follows:

`blockBackgroundInteraction: true,`

If the user tries to interact while the SnackBar is visible, the SnackBar disappears. Then, they can close the bottomSheet as they wish.

I don't know why the provider didn't dispose, and maybe this isn't the best solution. But if someone experiences the same problem and comes across this topic later, they can take such a solution path.
